services/QuillLMS/app/models/schools_users.rb

```diff
@@ -20,8 +20,25 @@ class SchoolsUsers < ApplicationRecord
 
   # When a teacher sets their school, we make sure they they have the appropriate subscription type.
   after_save :update_subscriptions
+  after_save :school_changed_change_log, if: :should_write_change_log?
 
   def update_subscriptions
     user&.updated_school(school_id)
   end
+
+  private def should_write_change_log?
+    (saved_change_to_attribute?(:school_id) &&
+      attribute_before_last_save(:school_id))
+  end
+
+  private def school_changed_change_log
+    ChangeLog.create({
+      action: ChangeLog::USER_ACTIONS[:update],
+      changed_record_type: 'User',
```

Comment:
You are using two strings to link logic between here and the run (`User` and `school`), it would be better if they referenced constants.